src/rc_string.c

```diff
@@ -65,29 +66,41 @@ static void rc_string_dispose(RC_STRING* content)
 
 static THANDLE(RC_STRING) rc_string_create_impl(const char* string)
 {
+    THANDLE(RC_STRING) result = NULL;
+
     /* Codes_SRS_RC_STRING_01_002: [ Otherwise, rc_string_create shall determine the length of string. ]*/
     size_t string_length = strlen(string);
-    size_t string_length_with_terminator = string_length + 1;
-
-    /* Codes_SRS_RC_STRING_01_003: [ rc_string_create shall allocate memory for the THANDLE(RC_STRING), ensuring all the bytes in string can be copied (including the zero terminator). ]*/
-    THANDLE(RC_STRING) temp_result = THANDLE_MALLOC_WITH_EXTRA_SIZE(RC_STRING)(rc_string_dispose, sizeof(RC_STRING_INTERNAL) - sizeof(RC_STRING) + string_length_with_terminator);
-    if (temp_result == NULL)
+    size_t memory_allocation_max_size = SIZE_MAX - (sizeof(RC_STRING_INTERNAL) - sizeof(RC_STRING) + 1);
```

Comment:
> + 1 [](http://example.com/codeflow?start=98&length=4)

Are you sure this + 1 is correct here?

Reply:
Ah I see, this is the +1 because we want the null terminator.
It works as is but consider changing the check in line to be

    if (string_length_with_terminator > SIZE_MAX - (sizeof(RC_STRING_INTERNAL) - sizeof(RC_STRING)))

Looks cleaner than having the +1 as it is now. And we need to compute string_length_with_terminator anyway for later computations